AdminCore.FsmWorkflow/FsmMachines/WorkflowFsmPto.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Generic;
+using AdminCore.Common;
+using AdminCore.Constants.Enums;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveStates;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveTriggers;
+using AdminCore.FsmWorkflow.FsmMachines.FsmWorkflowState;
+using Stateless;
+
+namespace AdminCore.FsmWorkflow.FsmMachines
+{
+    public class WorkflowFsmPto : WorkflowFsm<PtoState, LeaveTriggersPto>
+    {       
+        private StateMachine<PtoState, LeaveTriggersPto>.TriggerWithParameters<EventStatuses, string> LeaveResponseTrigger;
+        
+        public WorkflowFsmPto(WorkflowStateData fsmStateData)
+        {
+            ConfigureFsm(fsmStateData);
+        }
+        
+        public override void ConfigureFsm(WorkflowStateData fsmStateData)
+        {
+            FsmStateData = fsmStateData;
+            FsMachine = new StateMachine<PtoState, LeaveTriggersPto>(() => (PtoState)FsmStateData.CurrentState, s => FsmStateData.CurrentState = (int)s);
+            
+            LeaveResponseTrigger = FsMachine.SetTriggerParameters<EventStatuses, string>(LeaveTriggersPto.LeaveResponded);
+            
+            // Leave Awaiting Team Lead and Client
+            FsMachine.Configure(PtoState.LeaveAwaitingTeamLeadClient)
+                .SubstateOf(PtoState.LeaveAwaitingResponses)
+                .OnActivate(() =>
+                {
+                    if (IsAdminResponseReceived(FsmStateData.ApprovalDict))
+                    {
+                        FsMachine.Fire(IsAdminResponseApprove(FsmStateData.ApprovalDict)
+                            ? LeaveTriggersPto.AdminApprove
+                            : LeaveTriggersPto.AdminReject);
+                    }
+                    else if (IsTeamLeadClientResponsesReceived(FsmStateData.ApprovalDict))
```

Comment:
can we get into a scenario when admin/team lead response is received? if such case can occur what happens then?

Reply:
As discussed offline, it will NoOp if an incorrect role is used to fire a response.